lib/matplotlib/axes/_axes.py

```diff
@@ -3112,6 +3116,74 @@ def boxplot(self, x, notch=False, sym=None, vert=True, whis=1.5,
                             if ci[1] is not None:
                                 stats['cihi'] = ci[1]
 
+        if notch is None:
+            notch = rcParams['boxplot.notch']
+        if vert is None:
+            vert = rcParams['boxplot.vertical']
+        if patch_artist is None:
+            patch_artist = rcParams['boxplot.patchartist']
+        if meanline is None:
+            meanline = rcParams['boxplot.meanline']
+        if showmeans is None:
+            showmeans = rcParams['boxplot.showmean']
+        if showcaps is None:
+            showcaps = rcParams['boxplot.showcaps']
+        if showbox is None:
+            showbox = rcParams['boxplot.showbox']
+        if showfliers is None:
+            showfliers = rcParams['boxplot.showfliers']
+
+        if flierprops.get('color') == 'b':
+            flierprops['color'] = rcParams['boxplot.flierprops.color']
+        if flierprops.get('marker') == '+':
+            flierprops['marker'] = rcParams['boxplot.flierprops.marker']
+        if not flierprops.get('markerfacecolor', None):
+            flierprops['markerfacecolor'] = rcParams[('boxplot.flierprops.'
+                                                        'markerfacecolor')]
+        if not flierprops.get('markeredgecolor', None):
+            flierprops['markeredgecolor'] = rcParams[('boxplot.flierprops.'
+                                                        'markeredgecolor')]
+        if not flierprops.get('markersize', None):
+            flierprops['markersize'] = rcParams[('boxplot.flierprops.'
+                                                    'markersize')]
+        if not flierprops.get('linestyle', None):
+            flierprops['linestyle'] = rcParams[('boxplot.flierprops.'
+                                                            'linestyle')]
+
+        if not boxprops:
```

Comment:
Do you mean to say that I should put all the props dict to default to an empty dict on function call, and then do something like : 
```python
if boxplot :
    # rc Parameters
```

Also, I just tought that I should always create the dictionnaries with the rcParameters and update them afterwards with the passed dictionnaries. If I don't, it would ignore any parameter from the rc even if we pass a dict with a single value. In that case, it would change the way of doing it quite a lot.